src/inspector_js_api.cc

```diff
@@ -32,6 +32,162 @@ std::unique_ptr<StringBuffer> ToProtocolString(Isolate* isolate,
   return StringBuffer::create(StringView(*buffer, buffer.length()));
 }
 
+class JSDispatcher;
+
+class JSDispatcherInterface : private AsyncWrap {
+ public:
+  JSDispatcherInterface(Environment* env, Local<Object> object)
+                      : AsyncWrap(env, object, PROVIDER_INSPECTORJSDISPATCHER),
+                        env_(env),
+                        dispatcher_(nullptr) {
+    Wrap(object, this);
+    env->SetMethod(object, "sendMessageToFrontend",
+                   JSDispatcherInterface::SendMessageToFrontend);
+  }
+
+  ~JSDispatcherInterface() override;
+
+  bool HandleMessage(const v8_inspector::StringView& message) {
+    Isolate* isolate = env_->isolate();
+    HandleScope handle_scope(isolate);
+    if (env_->context().IsEmpty()) {
+      // Post-shutdown cleanup, should not be any real remaining messages
+      CHECK_EQ(message.length(), 0);
+      return false;
+    }
+    Context::Scope context_scope(env_->context());
+    Local<Value> argument;
+    if (message.length() > 0) {
+      MaybeLocal<String> v8string =
+          String::NewFromTwoByte(isolate, message.characters16(),
```

Comment:
Why is `message.is8Bit()` always false? If it is, can you add a comment explaining why + a `CHECK` for that?

Reply:
At this point, it just `// Should never happen` - but I added a second branch just in case.